src/javascript/Validation/validation.utils.js

```diff
@@ -35,38 +35,44 @@ export const onServerError = (error, formikActions, notificationContext, t, defa
                 formikActions.setFieldTouched('ce:systemName', true, false);
-                return;
             }
 
-            if (graphQLError.message &&
-                graphQLError.message.startsWith('org.jahia.services.content.PropertyConstraintViolationException') &&
-                graphQLError.message.includes('Invalid link')) {
-                // Custom handling for invalid link error
+            if (graphQLError.errorType === 'GqlConstraintViolationException' &&
+                graphQLError.extensions && graphQLError.extensions.constraintViolations &&
+                graphQLError.extensions.constraintViolations.length > 0) {
+                // Constraint violation errors
 
-                const extractedErrorData = graphQLError.message.match(/^org.jahia.services.content.PropertyConstraintViolationException: (?<prop>(.*?)): Invalid link(?<link>(.*?))$/);
-                if (extractedErrorData && extractedErrorData.groups && extractedErrorData.groups.prop && extractedErrorData.groups.link) {
-                    const propNameIndex = extractedErrorData.groups.prop.lastIndexOf(' ');
-                    const propName = extractedErrorData.groups.prop.substring(propNameIndex + 1);
+                for (const constraintViolation of graphQLError.extensions.constraintViolations) {
+                    console.log('Constraint violation error: ', constraintViolation);
+                    if (constraintViolation.propertyName) {
+                        if (constraintViolation.constraintMessage.startsWith('Invalid link')) {
+                            // Custom handling for invalid link error
+                            formikActions.setFieldError(constraintViolation.propertyName, 'invalidLink_' + constraintViolation.constraintMessage.substring('Invalid link'.length));
+                        } else {
+                            // Default constraint violation handling
+                            formikActions.setFieldError(constraintViolation.propertyName, 'constraintViolation_' + constraintViolation.constraintMessage);
+                        }
 
-                    notificationContext.notify(t('content-editor:label.contentEditor.error.invalidLinks'), ['closeButton']);
-                    formikActions.setFieldError(propName, 'invalidLink_' + extractedErrorData.groups.link);
-                    formikActions.setFieldTouched(propName, true, false);
-                    return;
+                        formikActions.setFieldTouched(constraintViolation.propertyName, true, false);
+                        notificationErrorMessage = t('content-editor:label.contentEditor.error.constraintViolations');
+                    }
                 }
             }
         }
     }
 
-    // No error handling, print error in console and use default error message
-    console.error(error);
-    notificationContext.notify(t(defaultErrorMessage), ['closeButton']);
+    if (notificationErrorMessage) {
```

Comment:
we will have always default error message, no?

Reply:
not in case of javax.jcr.ItemExistsException
the error is: content-editor:label.contentEditor.error.changeSystemName

Reply:
and in case of constraints violations:
'content-editor:label.contentEditor.error.constraintViolations'

Reply:
I'm talking about this one:

let notificationErrorMessage = t(defaultErrorMessage);